integration-test/1810-alternate-viewpoints.py

```diff
@@ -0,0 +1,153 @@
+        #
+        self.generate_fixtures(
+            # "a" relation gives us a linestring.
+            dsl.way(1, dsl.fit_in_tile(
+                z, x, y, 'LINESTRING(0.1 0.5, 0.1 0.9, 0.5 0.9)'), {
+                    'admin_level': '2',
+                    'boundary': 'claim',
+                    'claimed_by': 'XX',
+                    'source': 'openstreetmap.org',
+                    'type': 'boundary',
+                }),
+            # "b" ways give us linestring(s)
+            dsl.way(2, dsl.fit_in_tile(
+                z, x, y, 'LINESTRING(0.1 0.5, 0.5 0.5, 0.5 0.9)'), {
+                    'admin_level': '2',
+                    'boundary': 'administrative',
+                    'disputed': 'yes',
+                    'disputed_by': 'XX',
+                    'source': 'openstreetmap.org',
+                }),
+            # "b & c" ways + country relation give us a polygon => oriented
+            # boundary curve. we get an oriented boundary curve for each
+            # country boundary.
+            dsl.way(3, dsl.fit_in_tile(
+                z, x, y,
+                'LINESTRING(0.5 0.9, 0.9 0.9, 0.9 0.1, 0.1 0.1, 0.1 0.5)'), {
+                    'admin_level': '2',
+                    'boundary': 'administrative',
+                    'name': 'XX',
+                    'source': 'openstreetmap.org',
+                    'mz_boundary_from_polygon': True,  # need this for hack
+                }),
+            dsl.way(4, dsl.fit_in_tile(
+                z, x, y,
+                'LINESTRING(0.1 0.5, 0.1 0.1, 0.9 0.1, 0.9 0.9, 0.5 0.9)'), {
+                    'admin_level': '2',
+                    'boundary': 'administrative',
+                    'name': 'YY',
+                    'source': 'openstreetmap.org',
+                    'mz_boundary_from_polygon': True,  # need this for hack
+                }),
+            # this is just here to turn off maritime boundaries for everything
+            # in this tile.
+            dsl.way(5, dsl.tile_box(z, x, y), {
+                'source': 'tilezen.org',
+                'maritime_boundary': True,
+                'min_zoom': 0,
+                'kind': 'maritime',
+            }),
+        )
+
+        # should get a non-disputed section of border
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'country',
+                'name:left': 'XX',
+                'name:right': 'YY',
+            })
+
+        # should get a section recognised only by XX
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'unrecognized_country',
+                'kind:xx': 'country',
+            })
+
+        # should get a section recognised _except_ by XX
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'country',
+                'kind:xx': 'unrecognized_country',
```

Comment:
Please also add a `name` test?

Reply:
Added `name` test in 67432dc4 (and also fixed bug in test).